### PR TITLE
Loosen SmartyStreets Validation

### LINKED ISSUE #1230: Adjust sensitivity of SmartyStreets address validation
The sensitivity of address validation is slowing users down, and in some cases is presenting a suggested address even when the user-entered address is exactly the same. Can we adjust the sensitivity level? 

Background: A user reported that SmartyStreets is flagging addresses and suggesting the exact same address as an alternative to what the user inputted. Note: We did not see this in action and suspect that SS is just very sensitive to variations - e.g. ST vs. Street. The user may not be noticing these subtleties. 

We later tried to replicate this and found that even with an exact match, the address validation still appears (see screenshot below).

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- Resolves #1230

## Changes Proposed

- Adds a relatively blunt helper function, `suggestionIsCloseEnough`, and logic to just _accept_ the SmartyStreets suggestion if it's close enough

## Additional Information

- Close enough, here, is defined as "case and whitespace insensitive string equality," and ignores extended +4 zip code additions. Changes or additions to County are also not prompted.

### UI
- [x] Any changes to the UI/UX have been approved by design 
- [x] Any new or edited error messages have been approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support

### CODE REVIEW

File: frontend/src/app/utils/smartyStreets.ts [hunk 1]
```diff
@@ -55,3 +55,34 @@ export const getBestSuggestion = async (
...
+    return false;
+  }
+
+  const fields: (keyof Address)[] = ["city", "state", "street", "streetTwo"];
+  fields.forEach((field) => {
+    if (
+      toLowerStripWhitespace(original[field]) !==
+      toLowerStripWhitespace(suggested[field])
+    ) {
+      return false;
```
Comment (jeskew-gov) [inline frontend/src/app/utils/smartyStreets.ts:80]: `forEach` doesn't do anything with the return value, does it? I think the intent here was a `for` loop

Comment (nickscialli-usds) [reply frontend/src/app/utils/smartyStreets.ts:80]: A `some` or `every` loop might work here as well, but @jeskew-gov is correctly pointing out that the `forEach` loop is currently not affecting the return value of the `suggestionIsCloseEnough` function

Comment (nullflux) [reply frontend/src/app/utils/smartyStreets.ts:80]: Oh, yeah, duh. Thanks!

Fixed in 93e7b241

**Review by jeskew-gov — APPROVED** [review]

File: frontend/src/app/utils/smartyStreets.ts [hunk 2]
```diff
@@ -55,3 +55,34 @@ export const getBestSuggestion = async (
     console.error("Unable to validate address:", error.message);
   }
 };
+
+function toLowerStripWhitespace(s: string | null): string {
```
Comment (ClydeDroid) [inline frontend/src/app/utils/smartyStreets.ts:59]: Kind of a nitpick, but I think this function should go into `frontend/src/app/utils/text.ts`

**Review by ClydeDroid — APPROVED** [review]
Looks good!

File: frontend/src/app/utils/smartyStreets.ts (hunk 2 above)
Comment (nullflux) [reply frontend/src/app/utils/smartyStreets.ts]: Done in 6b18878c

**Pull request merged**